Scenes/map/map_room.gd

```diff
@@ -47,11 +47,24 @@ func _on_input_event(_viewport: Node, event: InputEvent, _shape_idx: int):
 	if not available or not event.is_action_pressed("left_mouse"):
 		return
 
-	print("input fired")
 	room.selected = true
 	animation_player.play("select")
 
 
 # called by the AnimationPlayer when the select animation finishes
 func _on_map_room_selected() -> void:
 	selected.emit(room)
+
+
+func enter_room() -> void:
+	room.selected = true
+	animation_player.play("select")
+
+
+func _on_body_entered(_body: Node2D):
```

Comment:
FYI I got stuck here because I was using on_area_entered as a signal and it wasn't working. Because the player is a CharacterBody2d node, you have to use on_body_entered signal